### PR TITLE
SQL: Fix issue with wrong data type for scripted Grouping keys

### PR DESCRIPTION
**Pull request by matriv**
When the grouping key of a GROUP BY is a painless script (functions are
involved), the data type of the key was incorrect in certain cases
(Boolean, IP, Date). This resulted in returning wrong data type for this
columns in the query results. E.g.:

```
SELECT COUNT(*), a > 10 AS a FROM t GROUP BY a
```

Fixes: #35662

### CODE REVIEW

File: x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/querydsl/agg/GroupByScriptKey.java [hunk 1]
```diff
@@ -39,12 +40,18 @@ public TermsValuesSourceBuilder asValueSource() {
                 .order(direction().asOrder())
                 .missingBucket(true);
 
-        if (script.outputType().isInteger()) {
+        if (script.outputType() == DataType.BOOLEAN) {
```
Comment (costin) [inline x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/querydsl/agg/GroupByScriptKey.java]: Why pick this order? Either is alphabetically or usage, in which case, boolean should be at the bottom along with IP and the rest higher up.

**Review by costin — APPROVED** [review]
LGTM

**Review by astefan — APPROVED** [review]
LGTM

**Comment by matriv** [comment]
Backported to `6.x` with 0352b01b976c8d20c7810db44070dc15cc358572

**Comment by matriv** [comment]
Backported to `6.5` with c4f71d799fdc6f33ccfa98e360e12f48bc3a122f

**Pull request merged**